src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -138,8 +138,7 @@ public void updateIsFailed(UUID id) {
     }
 
     public void checkPrintState() {
-        // TODO
-        letterRepository.getStaleLetters();
+        letterRepository.getStaleLetters().forEach(insights::trackNotPrintedLetter);
```

Comment:
I have a query here what happens when say there are multiple letters which are not printed and `telemetry.trackEvent` fails for say first one?Do we need a try catch in `trackNotPrintedLetter(..)` method?

Reply:
Looked into client and the only way it will throw the error is if instrumentation key is empty. We are making sure in logging library that it is not. Otherwise client catches all throwables and logs an error with built-in logger